scripts/S3_FSCK/s3_fsck_p2_reverselookup.py

```diff
@@ -0,0 +1,163 @@
+SPLITGETUSERMD_BIN = cfg["nasdk_tools"]["scalsplitgetusermd_bin"]
+PORT_RANGE = cfg["nasdk_tools"]["port_range"]
+ARCDATA_DRIVER = cfg["nasdk_tools"]["arcdata_path"]
+
+
+os.environ["PYSPARK_SUBMIT_ARGS"] = '--packages "org.apache.hadoop:hadoop-aws:2.7.3" pyspark-shell'
+spark = SparkSession.builder \
+     .appName("s3_fsck_p2_reverselookup.py: lookup the input keys for the newly found orphaned DATA RING main chunks. If their service ID is SPLIT, look the top input key up. RING: " + RING) \
+     .config("spark.hadoop.fs.s3a.impl", "org.apache.hadoop.fs.s3a.S3AFileSystem")\
+     .config("spark.hadoop.fs.s3a.access.key", ACCESS_KEY)\
+     .config("spark.hadoop.fs.s3a.secret.key", SECRET_KEY)\
+     .config("spark.hadoop.fs.s3a.endpoint", ENDPOINT_URL) \
+     .config("spark.executor.instances", cfg["spark.executor.instances"]) \
+     .config("spark.executor.memory", cfg["spark.executor.memory"]) \
+     .config("spark.executor.cores", cfg["spark.executor.cores"]) \
+     .config("spark.driver.memory", cfg["spark.driver.memory"]) \
+     .config("spark.memory.offHeap.enabled", cfg["spark.memory.offHeap.enabled"]) \
+     .config("spark.memory.offHeap.size", cfg["spark.memory.offHeap.size"]) \
+     .config("spark.local.dir", cfg["path"]) \
+     .getOrCreate()
+
+#spark.sparkContext.setLogLevel("DEBUG")
+#print(os.environ["PATH"])
+input_path = "%s://%s/%s/s3fsck/s3objects-missing.csv" % (PROTOCOL, PATH, RING)
+output_path = "%s://%s/%s/s3fsck/reverselookup.csv" % (PROTOCOL, PATH, RING)
+
+print("REVLOOKUP_BIN:", REVLOOKUP_BIN)
+print("SPLITGETUSERMD_BIN:", SPLITGETUSERMD_BIN)
+print("PORT_RANGE:", PORT_RANGE)
+print("ARCDATA_DRIVER:", ARCDATA_DRIVER)
+def get_local_ip():
+    try:
+        output = subprocess.check_output("hostname -i", shell=True, stderr=open(os.devnull, 'wb')).decode().strip()
+        return output
+    except subprocess.CalledProcessError as e:
+        return "ERR_IP"
+
+def get_random_port():
+    return random.randint(PORT_RANGE[0], PORT_RANGE[1])
+
+def rev_lookup(key):
+    try:
+        ip_port = "{}:{}".format(get_local_ip(), get_random_port())
+        cmd = "ssh -qT $(hostname -i) {} -b '{}' {}".format(REVLOOKUP_BIN, ip_port, key)
+        print("  Executing rev_lookup command: {}".format(cmd))
+        output = subprocess.check_output(cmd, shell=True, stderr=open(os.devnull, 'wb'))
+        print("  rev_lookup output: {}".format(output))
+        if output:
+            return output.decode().strip()
+        else:
+            return "NULL"
+    except subprocess.CalledProcessError as e:
+        print("  rev_lookup error: {}".format(e.output.decode()))
+        return "ERR_REVLOOKUP"
+
+def split_lookup(key):
+    try:
+        ip_port = "{}:{}".format(get_local_ip(), get_random_port())
+        cmd = "ssh -qT $(hostname -i) {} -b '{}' -d '{}' {} 2>/dev/null | strings | awk '/index/{{$1=substr($1,length($1)-39,40); print}}'".format(SPLITGETUSERMD_BIN, ip_port, ARCDATA_DRIVER, key)
```

Comment:
```suggestion
        cmd = "ssh -qT $(hostname -i) {} -b '{}' -d '{}' {} 2>/dev/null | strings -t binary -a | awk '/index/{{$1=substr($1,length($1)-39,40); print}}'".format(SPLITGETUSERMD_BIN, ip_port, ARCDATA_DRIVER, key)
```
otherwise strings may try to be too smart and do weird stuff

Reply:
Hmm, thanks for the heads up. I've just tried it and the shell complains: 

```bash
scalsplitgetusermd -b 10.4.12.21:4244 -d arcdata 0014CF2744580F1F0AA8DF00000000511470C070 | strings -t binary -a
Usage: strings [option(s)] [file(s)]
 Display printable strings in [file(s)] (stdin by default)
 The options are:
  -a - --all                Scan the entire file, not just the data section [default]
  -d --data                 Only scan the data sections in the file
  -f --print-file-name      Print the name of the file before each string
  -n --bytes=[number]       Locate & print any NUL-terminated sequence of at
  -<number>                   least [number] characters (default 4).
  -t --radix={o,d,x}        Print the location of the string in base 8, 10 or 16
  -w --include-all-whitespace Include all whitespace as valid string characters
  -o                        An alias for --radix=o
  -T --target=<BFDNAME>     Specify the binary file format
  -e --encoding={s,S,b,l,B,L} Select character size and endianness:
                            s = 7-bit, S = 8-bit, {b,l} = 16-bit, {B,L} = 32-bit
  --unicode={default|show|invalid|hex|escape|highlight}
  -U {d|s|i|x|e|h}          Specify how to treat UTF-8 encoded unicode characters
  -s --output-separator=<string> String used to separate strings in output.
  @<file>                   Read options from <file>
  -h --help                 Display this information
  -v -V --version           Print the program's version number
strings: supported targets: elf64-x86-64 elf32-i386 elf32-iamcu elf32-x86-64 a.out-i386-linux pei-i386 pei-x86-64 elf64-l1om elf64-k1om elf64-little elf64-big elf32-little elf32-big plugin srec symbolsrec verilog tekhex binary ihex
write: Broken pipe
write usermd failed
[Exit 1]
```

Reply:
Maybe you meant `-T binary` instead :)